src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests_ListPatterns.cs

```diff
@@ -673,6 +673,146 @@ public void LengthPattern_16()
             EOF();
         }
 
+        [Fact]
+        public void LengthPattern_17()
+        {
+            UsingExpression(@"c is not [<0]");
```

Comment:
Let's add full tests for those scenarios that now pass (verifying diagnostics and execution).